### PR TITLE
[16.0][MIG] procurement_purchase_no_grouping: Migration to 16.0

### PR DESCRIPTION
**Pull request by victoralmau**
Superseed https://github.com/OCA/purchase-workflow/pull/1877

Migration to 16.0

Please @pedrobaeza and @sergio-teruel can you review it?

@Tecnativa TT46588

### CODE REVIEW

File: procurement_purchase_no_grouping/models/stock_rule.py [hunk 1]
```diff
@@ -15,9 +15,10 @@ class StockRule(models.Model):
...
     def _run_buy(self, procurements):
         for procurement, _rule in procurements:
-            grouping = procurement.product_id.categ_id.procured_purchase_grouping
-            if not grouping:
-                grouping = self.env.company.procured_purchase_grouping
+            grouping = (
+                procurement.product_id.categ_id.procured_purchase_grouping
+                or self.env.company.procured_purchase_grouping
+            )
             procurement.values["grouping"] = grouping
```
Comment (pedrobaeza) [inline procurement_purchase_no_grouping/models/stock_rule.py]: And if you want to save a line in the code:

```suggestion
            procurement.values["grouping"] = (
                procurement.product_id.categ_id.procured_purchase_grouping
                or self.env.company.procured_purchase_grouping
            )

**Review by pedrobaeza — COMMENTED** [review]
I think both migration commits can be joined and put co-authorship in the resulting one, as it doesn't add too much value separated.

**Comment by victoralmau** [comment]
Changes done.

**Review by pedrobaeza — APPROVED** [review]

File: procurement_purchase_no_grouping/models/purchase_order.py [hunk 2]
```diff
@@ -29,10 +29,11 @@ def _find_candidate(
         modules, and on standard operation mode, procurement groups are not reused
         between sales orders.
         """
-        obj = self
         if values.get("grouping") == "line":
-            obj = self.filtered(lambda x: x.order_id.group_id == values.get("group_id"))
-        return super(PurchaseOrderLine, obj)._find_candidate(
+            self = self.filtered(
+                lambda x: x.order_id.group_id == values.get("group_id")
```
Comment (sergio-teruel) [inline procurement_purchase_no_grouping/models/purchase_order.py:34]: values.get("group_id") is an id or recordset??

Comment (victoralmau) [reply procurement_purchase_no_grouping/models/purchase_order.py:34]: It is a recordset (https://github.com/odoo/odoo/blob/16.0/addons/stock/models/stock_move.py#L1405)

Comment (pedrobaeza) [reply procurement_purchase_no_grouping/models/purchase_order.py:34]: Odoo putting very badly the variable names...

**Review by sergio-teruel — APPROVED** [review]

**Pull request merged**